### PR TITLE
Add webp support

### PR DESCRIPTION
**Pull request by tobiichi3227**
I added the webp format support.
It's my first PR for this project.
Thanks.

### CODE REVIEW

File: src/helpers/Webp.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    if (WebPDecode((const unsigned char*)imageRawData, fileInfo.st_size, &config) != VP8_STATUS_OK) {
+        Debug::log(CRIT, "createSurfaceFromWEBP: WebP Decode Failed (?)");
+        exit(1);
+    }
+
+    cairo_surface_mark_dirty(cairoSurface);
+    cairo_surface_set_mime_data(cairoSurface, CAIRO_MIME_TYPE_PNG, (const unsigned char*)imageRawData, fileInfo.st_size, free, imageRawData);
+
+	WebPFreeDecBuffer(&config.output);
```
Comment (vaxerski) [inline src/helpers/Webp.cpp:78]: indent oops

File: src/helpers/Webp.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include "../defines.hpp"
+#include <webp/decode.h>
```
Comment (vaxerski) [inline src/helpers/Webp.hpp:4]: no point in including webp here, can be done in the source file.

Comment (tobiichi3227) [reply src/helpers/Webp.hpp:4]: Ok, I will move it to source file.

File: src/helpers/Webp.cpp (hunk 1 above)
Comment (tobiichi3227) [reply src/helpers/Webp.cpp:78]: Oh, I'm sorry.
It's is a tab.

File: src/helpers/Webp.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    auto cairoSurface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, width, height);
+    if (cairo_surface_status(cairoSurface) != CAIRO_STATUS_SUCCESS) {
+        Debug::log(CRIT, "createSurfaceFromWEBP: Cairo Failed (?)");
+        cairo_surface_destroy(cairoSurface);
+        exit(1);
+    }
+
+
+    if (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__) {
```
Comment (vaxerski) [inline src/helpers/Webp.cpp:55]: no {} around short ifs

File: src/helpers/Webp.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        exit(1);
+    }
+
+    if (WebPGetFeatures((const unsigned char*)imageRawData, fileInfo.st_size, &config.input) != VP8_STATUS_OK) {
+        Debug::log(ERR, "createSurfaceFromWEBP: file is not webp format");
+        free(imageRawData);
+        exit(1);
+    }
+
+    const auto height = config.input.height;
```
Comment (vaxerski) [inline src/helpers/Webp.cpp:44]: consts are CAPS

**Review by vaxerski — COMMENTED** [review]
apart from these lgtm

**Review by vaxerski — APPROVED** [review]
g

**Comment by vaxerski** [comment]
@fufexan nix faisl cuz no dep

**Comment by vaxerski** [comment]
top

**Comment by vaxerski** [comment]
thanks!

**Pull request merged**